units/event_metajob.go

```diff
@@ -302,19 +146,26 @@ func (j *eventMetaJob) Run(ctx context.Context) {
 	if j.limit <= 0 {
 		j.limit = evergreen.DefaultEventProcessingLimit
 	}
-	j.events, err = event.FindUnprocessedEvents(j.limit)
+	j.events, err = event.FindUnprocessedEvents(-1)
 	if err != nil {
 		j.AddError(err)
 		return
 	}
+	grip.Info(j.events)
 
 	j.AddError(errors.Wrap(j.logEventCount(), "can't log unprocessed event count"))
 
 	if len(j.events) == 0 {
 		return
 	}
-
-	j.AddError(j.dispatchLoop(ctx))
+	eventIDs := []string{}
+	for i, evt := range j.events {
+		eventIDs = append(eventIDs, evt.ID)
+		if (i+1)%j.limit == 0 || i+1 == len(j.events) {
+			j.q.Put(ctx, NewEventNotifierJob(j.env, j.q, sha256sum(eventIDs), eventIDs))
```

Comment:
I might just pass eventIDs here, and put the hashing in the constructor for the notifier job.